src/doc/Document.ts

```diff
@@ -41,7 +42,7 @@ export declare namespace Document {
   }
 }
 
-export class Document<T = unknown> {
+export class Document<T extends Node = Node> {
```

Comment:
I would _really_ like for `new Document(foo)` to automatically type itself as `Document<NodeType<typeof foo>>`, but I can't figure out how to make that happen.